core/launch.go

```diff
@@ -0,0 +1,225 @@
+package core
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"chain/core/accesstoken"
+	"chain/core/account"
+	"chain/core/asset"
+	"chain/core/config"
+	"chain/core/fetch"
+	"chain/core/generator"
+	"chain/core/leader"
+	"chain/core/pin"
+	"chain/core/query"
+	"chain/core/rpc"
+	"chain/core/txbuilder"
+	"chain/core/txdb"
+	"chain/core/txfeed"
+	"chain/database/pg"
+	"chain/log"
+	"chain/protocol"
+	"chain/protocol/bc"
+)
+
+const (
+	blockPeriod              = time.Second
+	expireReservationsPeriod = time.Second
+)
+
+// LaunchOption describes a runtime configuration option.
+type LaunchOption func(*API)
+
+// AlternateAuth configures the Core to use authFn to authenticate
+// incoming requests in addition to the default access token authentication.
+func AlternateAuth(authFn func(*http.Request) bool) LaunchOption {
+	return func(a *API) { a.altAuth = authFn }
```

Comment:
Based on my reading of API.altAuth and apiAuthn.alt, currently this LaunchOption is actually required yeah? If the caller didn't provide it, I think the core would crash.

I think if we're using this pattern it should be a rule that anything that's an "option" needs to be optional, as in, you can leave it out and things still work.